src/redux/actions/cellSets/createCellSet.js

```diff
@@ -17,25 +39,58 @@ const createCellSet = (experimentId, name, color, cellIds) => async (dispatch, g
     key: uuidv4(),
     name,
     color,
-    cellIds: new Set([...cellIds].map((id) => parseInt(id, 10))),
     type: 'cellSets',
   };
 
-  if (data.cellIds.size === 0) {
+  // cellIds are stored as an integer set in redux, so we need to convert it
+  const dataForRedux = {
+    ...data,
+    cellIds: cellIdsSet,
+  };
+
+  // cellIds are stored as an array in S3
+  const dataForUpload = {
+    ...data,
+    cellIds: Array.from(cellIdsSet),
+  };
+
+  if (dataForRedux.cellIds.size === 0) {
     pushNotificationMessage('info', endUserMessages.EMPTY_CLUSTER_NOT_CREATED);
     return;
   }
 
   dispatch({
     type: CELL_SETS_CREATE,
-    payload: {
-      experimentId,
-      ...data,
-    },
+    payload: dataForRedux,
   });
 
-  await dispatch(saveCellSets(experimentId));
-  pushNotificationMessage('info', endUserMessages.NEW_CLUSTER_CREATED);
+  const url = `/v1/experiments/${experimentId}/cellSets`;
+
+  try {
+    const response = await fetchAPI(
+      url,
+      {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/boschni-json-merger+json',
```

Comment:
What made you use this application with the `CellSetJsonMerger` from the way it was before, is it faster?

Reply:
updateCellSets is uploading the whole cellsets object, this only performs the update we need to perform.

In some cases (experiment with a cellSets in S3 of 10 mb) before something like a cellset rename took very long.